### DISCUSSION TITLE
Compass lost every few seconds

### DISCUSSION THREAD

**Discussion by BloodSakura3774** [iNavFlight/inav#8791] [Q&A] [answered]
So, I got a m8q-5883 with QMC5883 mag, and it is connect to my fc sda scl correctly. My fc is iFlight Blitz F7. 

The problem is the mag signal to the fc loses randomly. And it is a very short lost, like a blink.

 I have tried to put it on another fc and everything is fine, I also tried to flash my fc to different version of iNav/BetaFlight, and it all behaved the same. 

I tried to change a pair of wire and it did improve the behavior —- it was connected to the fc like 5 seconds per minute and after I change the wire it is disconnected from the fc 5 seconds per minute.

 I’m 80% percent sure it is some sort of interference but I don’t have any other better solutions to that. Is someone able to provide me some help? Or if someone could tell me whether the coptor will crush during RTH/Failsafe and lose compass for a blink or it’s fine to lose it for such a short time.

**Comment by BloodSakura3774** [comment]
update, tried to change wire again and it’s better than last time but still kept disconnecting.

**Comment by DzikuVx** [comment, marked as answer]
It can be either an electrical interference or damaged compass chip even. Those kinds of problem are very hard to pinpoint without an oscilloscope.
Anyhow, the things to try:
1 - add additional pullup resistors for I2C lines
2 - use beter quality cables
3 - do a better soldering job
4 - twist the cables together
5 - make them shorter
6 - lower I2C speed in INAV

**Comment by BloodSakura3774** [comment]
Solved. After trying every solution provided by pawel and internet, nothing changed, it still disconnects. I then decided to try another flight controller and the problem is immediately solved, even with my “garbage” wire. So I assume I just had a faulty fc.
